src/features/auth/pages/signup/SetPassword/SetPassword.tsx

```diff
@@ -1,49 +1,57 @@
-import React, { FunctionComponent, useCallback, useRef, useState } from 'react'
-import { TextInput as RNTextInput } from 'react-native'
-import { v4 as uuidv4 } from 'uuid'
+import { yupResolver } from '@hookform/resolvers/yup'
+import React, { FunctionComponent, useCallback } from 'react'
+import { useForm } from 'react-hook-form'
 
-import {
-  isPasswordCorrect,
-  PasswordSecurityRules,
-} from 'features/auth/components/PasswordSecurityRules'
+import { setPasswordSchema } from 'features/auth/pages/signup/SetPassword/schema/setPasswordSchema'
 import { PreValidationSignupStepProps } from 'features/auth/types'
+import { PasswordInputController } from 'shared/forms/controllers/PasswordInputController'
 import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
 import { Form } from 'ui/components/Form'
-import { PasswordInput } from 'ui/components/inputs/PasswordInput'
 import { Spacer } from 'ui/theme'
 
-export const SetPassword: FunctionComponent<PreValidationSignupStepProps> = (props) => {
-  const [password, setPassword] = useState('')
-  const disabled = !isPasswordCorrect(password)
+type FormValues = {
+  password: string
+}
```

Comment:
il y a probablement moyen d'inférer le type depuis le schéma Yup

Reply:
Je me suis rappelé cet article que Pierre m'avait transféré : https://www.brendonovich.dev/blog/the-ultimate-form-abstraction j'ai l'impression qu'on a une solution pour ça mais ça demande un peu de travail (je le noterai bien dans le prochain ticket de react hook formisation parce que celui là c'est plutot celui de Matthias...)

Reply:
Dans la partie "A Typesafe Coincidence"